comet-java-client/src/main/java/ml/comet/experiment/Experiment.java

```diff
@@ -114,8 +114,9 @@
      * @param metricName The name for the metric to be logged
      * @param metricValue The new value for the metric.  If the values for a metric are plottable we will plot them
      * @param step The current step for this metric, this will set the given step for this experiment
+     * @param epoch The current epoch for this metric, this will set the given epoch for this experiment
      */
-    void logMetric(String metricName, Object metricValue, long step);
+    void logMetric(String metricName, Object metricValue, long step, long epoch);
```

Comment:
please make step and epoch as optional (Long) 
add method overload to allow logging with and without steps, with and without epoch

Reply:
@Nimrod007, at present there is 
```
    public void logMetric(String metricName, Object metricValue) 
```
that uses current step and epoch. And there are setters for step and epoch. Should I add separate methods like this?
```
    public void logMetricWithStep(String metricName, Object metricValue, long step) 
    public void logMetricWithEpoch(String metricName, Object metricValue, long epoch) 
```

Reply:
we want:
```
logMetric(String metricName, Object metricValue, Long step, Long epoch)
logMetric(String metricName, Object metricValue, Long step)
logMetric(String metricName, Object metricValue, Long epoch)
logMetric(String metricName, Object metricValue)
```
and overload 1 impl 
```
logMetric(String metricName, Object metricValue, Long step, Long epoch)
```

Reply:
Updated as following:
```
logMetric(String metricName, Object metricValue, Long step, Long epoch)
logMetric(String metricName, Object metricValue, Long step)
logMetric(String metricName, Object metricValue)
```